packages/theme/src/cli/utilities/theme-environment/theme-polling.ts

```diff
@@ -0,0 +1,89 @@
+import {Checksum, Theme, ThemeFileSystem} from '@shopify/cli-kit/node/themes/types'
+import {fetchChecksums, fetchThemeAsset} from '@shopify/cli-kit/node/themes/api'
+import {outputDebug, outputInfo} from '@shopify/cli-kit/node/output'
+import {AdminSession} from '@shopify/cli-kit/node/session'
+import {AbortError, FatalError} from '@shopify/cli-kit/node/error'
+import {renderWarning} from '@shopify/cli-kit/node/ui'
+
+const POLLING_INTERVAL = 3000
+
+export function pollThemeEditorChanges(
+  targetTheme: Theme,
+  session: AdminSession,
+  remoteChecksum: Checksum[],
+  localThemeFileSystem: ThemeFileSystem,
+) {
+  outputDebug('Listening for changes in the theme editor')
+
+  return setTimeout(() => {
+    pollRemoteChanges(targetTheme, session, remoteChecksum, localThemeFileSystem)
+      .then((latestChecksums) => {
+        pollThemeEditorChanges(targetTheme, session, latestChecksums, localThemeFileSystem)
+      })
+      .catch((error) => {
+        if (error instanceof FatalError) {
+          throw error
+        }
+        renderWarning({body: `Error while checking for changes in the theme editor: ${error.message}`})
```

Comment:
I could use some input on this flow.

imo, we should be aborting on any FatalError but what I'm not 100 set on is whether **we should be retrying the sync or just aborting on any error**.

The biggest risk here is swallowing errors without visibility - to circumvent this, I'm rendering a warning to the users in this scenario


Some guidance is linked below. The guide mentions to avoid `try catching` AbortErrors, however since we are throwing any AbortErrors back up, I'm wondering if that applies here? _(AbortErrors are instances of FatalErrors)_

[Errors | cli](https://shopify.github.io/cli/cli-kit/errors.html#aborting-the-execution-using-errors)